src/KalturaUploaderReducer.ts

```diff
@@ -0,0 +1,24 @@
+import {
+  KalturaMediaEntry,
+  KalturaUploadToken,
+} from "kaltura-typescript-client/api/types";
+
+export type State =
+  | { id: "start" }
+  | { id: "media_uploaded"; uploadResult: KalturaUploadToken }
+  | { id: "complete"; entry: KalturaMediaEntry };
+
+export type Action =
+  | { id: "upload_successful"; uploadResult: KalturaUploadToken }
```

Comment:
hmmm, I sat here pondering this for awhile, even came up with some different words. But in the end, I think the key here is that this one is more so capturing state _transitions_ as opposed to direct _actions_. They still represent edges in the state graph though.

I wonder if perhaps I should simple rename `Action` to `Transition` to assist. What do you think? 

(Regarding use of `id` rather than `type` same reason as for `Status`.)

Reply:
I can understand the state transitions, but I think it's good to keep using `Action`.

Reply:
Okay, cool. Will leave as is.